### PR TITLE
run circuit in separate CI only when specific circuit/proof file change

### PR DESCRIPTION
**Pull request by samzkback**
- [x] fix circuit test cuz by proof API change
- [x] add circuit test in CI
- [x] CI run only when circuit code change and proof API change (as it takes long time to compile and generate verifier contract)
- [x] separate CI for circuit and contract,  as wasm/zkey changes in every circuit generate.

### CODE REVIEW

File: packages/circuits/tests/shuffle.ts [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
 import { readFileSync } from 'fs';
 const buildBabyjub = require('circomlibjs').buildBabyjub;
 const snarkjs = require('snarkjs');
 
-describe('Shuffle encrypt unit tests', function () {
+describe('Shuffle encrypt/decrypt benchmark tests', function () {
   const numBits = BigInt(251);
   const numCards = BigInt(52);
-  const numProfiling = 10;
+  const numProfiling = 1;
```
Comment (stechu) [inline packages/circuits/tests/shuffle.ts:14]: `3` to `5` times?

**Review by stechu — APPROVED** [review]

File: .github/workflows/circuit.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+# For more information see: https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages
+name: Node.js CI
+
+on:
+  pull_request:
+    branches:
+      - main                                  # when PR merge to main
+    paths :
+      - 'packages/circuits/**'                # when circuit file modified in PR
+      - 'packages/proof/**'                   # when proof file modified in PR
```
Comment (BoyuanFeng) [inline .github/workflows/circuit.yml:11]: should ci also run when `packages/contracts/**` or `jssdk` changes?

**Review by BoyuanFeng — CHANGES_REQUESTED** [review]

File: .github/workflows/circuit.yml (hunk 2 above)
Comment (samzkback) [reply .github/workflows/circuit.yml:11]: circuit test do not rely on contract/jssdk.

File: .github/workflows/circuit.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      #       ${{ runner.os }}-build-
+      #       ${{ runner.os }}-
+      # - if: ${{ steps.cache-circom.outputs.cache-hit != 'true' }}
+    
+      - name: Install latest nightly
+        uses: actions-rs/toolchain@v1
+        with:
+            toolchain: nightly
+            override: true
+            components: rustfmt, clippy
```
[the remark this answers is not in the document]
Comment (samzkback) [reply .github/workflows/circuit.yml]: Yeah, removed

File: .github/workflows/circuit.yml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      #   uses: actions/cache@v3
+      #   with:
+      #     # npm cache files are stored in `~/.npm` on Linux/macOS
+      #     path: ~/.npm
+      #     key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hash('11') }}
+      #     restore-keys: |
+      #       ${{ runner.os }}-build-${{ env.cache-name }}-
+      #       ${{ runner.os }}-build-
+      #       ${{ runner.os }}-
+      # - if: ${{ steps.cache-circom.outputs.cache-hit != 'true' }}
```
[the remark this answers is not in the document]
Comment (samzkback) [reply .github/workflows/circuit.yml]: Yeah, removed

**Review by BoyuanFeng — APPROVED** [review]

**Pull request merged**